modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/service/impl/CTCollectionLocalServiceImpl.java

```diff
@@ -857,6 +860,97 @@ public boolean isCTEntryEnclosed(
 		return true;
 	}
 
+	@Override
+	public void moveCTEntries(
+			long fromCTCollectionId, long toCTCollectionId,
+			List<CTEntry> ctEntries)
+		throws PortalException {
+
+		CTCollection fromCTCollection =
+			ctCollectionPersistence.findByPrimaryKey(fromCTCollectionId);
+
+		if ((fromCTCollection.getStatus() != WorkflowConstants.STATUS_DRAFT) &&
+			(fromCTCollection.getStatus() !=
+				WorkflowConstants.STATUS_EXPIRED) &&
+			(fromCTCollection.getStatus() !=
+				WorkflowConstants.STATUS_PENDING)) {
+
+			throw new CTCollectionStatusException(
+				"Change tracking collection " + fromCTCollection +
+					" is read only");
+		}
+
+		CTCollection toCTCollection = ctCollectionPersistence.findByPrimaryKey(
+			toCTCollectionId);
+
+		if (toCTCollection.isReadOnly()) {
+			throw new CTCollectionStatusException(
+				"Change tracking collection " + toCTCollection +
+					" is read only");
+		}
+
+		Map<Long, List<CTEntry>> relatedCTEntriesMap = new HashMap<>();
+
+		for (CTEntry ctEntry : ctEntries) {
+			if (!_isMovable(fromCTCollectionId, ctEntry)) {
+				return;
+			}
+
+			relatedCTEntriesMap.putAll(
```

Comment:
@brookedalton doesn't the putAll method just replace the entries instead merging the lists?

Reply:
https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#putAll-java.util.Map-